pkg/util/active_groups.go

```diff
@@ -91,11 +84,31 @@ func (ag *ActiveGroups) PurgeInactiveGroupsForUser(userID string, deadline int64
 	return deletedGroups
 }
 
+func (ag *ActiveGroups) PurgeInactiveGroups(inactiveTimeout time.Duration, cleanupFuncs ...func(string, string)) {
+	userIDs := make([]string, len(ag.timestampsPerUser))
+	ag.mu.RLock()
+	i := 0
+	for userID := range ag.timestampsPerUser {
+		userIDs[i] = userID
+		i++
+	}
+	ag.mu.RUnlock()
+
+	for _, userID := range userIDs {
+		inactiveGroups := ag.PurgeInactiveGroupsForUser(userID, time.Now().Add(-inactiveTimeout).UnixNano())
```

Comment:
Please compute `time.Now().Add(-inactiveTimeout).UnixNano()` just once, before the `for` loop and then just reuse it.